### PR TITLE
Support for pointer-to-slice parameters

### PR DESCRIPTION
**Pull request by ondra**
Implement support for pointer-to-slice parameters for methods and functions:

Python:
```
import tst
s = tst.StrSlice()
tst.Fill(s)
print s
```

Go:
```
package tst
type StrSlice []string
func Fill(s *StrSlice) {
        *s = StrSlice{"A", "B"}
}
```

This doesn't implement pointer support in general (as we didn't need it), but the fixes should make the implementation easier.

### CODE REVIEW

**Comment by ondra** [comment]
I'd like to support the other backends and py3k eventually, but this is what we're using now.

File: bind/gengo.go [hunk 1]
```diff
@@ -588,6 +603,10 @@ func (g *goGen) genType(sym *symbol) {
 	if sym.isBasic() && !sym.isNamed() {
 		return
 	}
+	if (sym.kind & skPointer) != 0 {
```
Comment (sbinet) [inline bind/gengo.go]: please remove this debugging leftover :)

File: bind/gencpy_type.go [hunk 2]
```diff
@@ -20,6 +20,9 @@ func (g *cpyGen) genType(sym *symbol) {
 	if sym.isBasic() && !sym.isNamed() {
 		return
 	}
+	if (sym.kind & skPointer) != 0 {
```
Comment (sbinet) [inline bind/gencpy_type.go]: couldn't this be rewritten as:
```go
if sym.isPointer() {
    return
}
```

**Comment by ondra** [comment]
How would you want the test to look like? Is it enough to do it through `_examples`?

I've been digging through the code trying to make pointers work better in general, many changes would be necessary, but I'm missing the bigger picture, so I don't think I can fix it now.

Are you willing to merge this without larger changes? I do _think_ the code is slightly better than before.

**Comment by sbinet** [comment]
Yes, testing via the usual _examples is what I had in mind.
And I agree, the code looks better.
Also yes: willing to merge this once there's a test :)

**Comment by ondra** [comment]
Great:)! The test is there.
I'll add support for CFFI later.
Thanks for the review, and for GoPy, of course.

**Pull request merged**